src/catchP.js

```diff
@@ -0,0 +1,38 @@
+var _curry2 = require('./internal/_curry2');
+var _isFunction = require('./internal/_isFunction');
+var toString = require('./toString');
+
+
+/**
+ * Returns the result of applying the onFailure function to the value inside
+ * a failed promise. This is useful for handling rejected promises
+ * inside function compositions.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig (b -> c) -> (Promise a b) -> (Promise c d)
+ * @sig (b -> (Promise c d)) -> (Promise a b) -> (Promise c d)
+ * @param {Function} onFailure The function to apply.  Can return a value or a promise.
+ * @param {Promise} p
+ * @return {Promise} The result of calling `p.then(onFailure)`
+ *
+ * @example
+ *
+ *      var failedFetch = (id) => Promise.reject('bad ID');
+ *      var useDefault = () => ({ firstName: 'Bob', lastName: 'Loblaw' })
+ *
+ *      //recoverFromFailure :: String -> Promise [{firstName, lastName}]
+ *      var recoverFromFailure = R.pipe(
+ *        failedFetch(12345),
+ *        R.catchP(useDefault),
+ *        R.then(R.pick(['firstName', 'lastName']))
+ *      );
+ */
+module.exports = _curry2(function catchP(f, p) {
```

Comment:
The name `catch` is already taken 😜 .  I am not sure on this name because it creates an asymmetry with `then`, but I am lacking a decent alternative.  I don't like using `R.onSuccess` and `R.onFailure` because it's less obvious the functions relate to Promises.